### PR TITLE
chore(breaking): update dependencies with new Babel parser

### PR DESCRIPTION
**Pull request by jaworek**
### Summary

Update all project dependencie to the newest version. 
Especially `babel-eslint` was migrated to the new package name `@babel/eslint-parser`.
Due to NextJS not supporting Eslint v8 it was not updated to that version yet.

### Test plan

Verify that eslint rules work correctly and there are no new unexpected rules are triggered.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "11.0.0",
   "description": "ESLint preset extending Airbnb, Flow, Prettier and Jest",
   "engines": {
-    "node": ">=10.13.0"
+    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
```
Comment (jaworek) [inline package.json:6]: We may consider also supporting non-LTS Node versions.

File: node.js [hunk 2]
```diff
@@ -44,9 +44,14 @@ module.exports = {
   overrides: [
     {
       files: ['*.js', '*.jsx'],
-      parser: 'babel-eslint',
+      parser: '@babel/eslint-parser',
+      parserOptions: {
+        'babelOptions': {
+          'presets': ['@babel/preset-react', '@babel/preset-flow']
```
Comment (thymikee) [inline node.js:50]: preset react should be included in `react.js`

File: package.json (hunk 1 above)
Comment (thymikee) [reply package.json:6]: Let's add `^13.0.0` and `^15.0.0`

File: react.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ module.exports = {
   extends: [
     require.resolve('./node.js'),
     'plugin:react/recommended',
-    'prettier/react',
+    'prettier',
```
Comment (thymikee) [inline react.js:9]: node.js already includes that

File: package.json (hunk 1 above)
Comment (thymikee) [reply package.json]: Updated based on https://nodejs.org/dist/latest-v16.x/docs/api/esm.html#modules-ecmascript-modules

**Comment by thymikee** [comment]
> Due to NextJS not supporting Eslint v8 it was not updated to that version yet

The config will still work with ESLint 8 though. As shown in the test app.

**Pull request merged**